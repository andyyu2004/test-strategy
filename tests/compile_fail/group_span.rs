use test_strategy::Arbitrary;

// A compile error needs to occur on line 6, not on line 4.
#[derive(Debug, Arbitrary)]
struct X {
    #[strategy((0u8..10u8))]
    x: u32,
}

fn main() {}
