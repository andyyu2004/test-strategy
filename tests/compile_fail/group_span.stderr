error[E0271]: type mismatch resolving `<Range<u8> as Strategy>::Value == u32`
 --> tests/compile_fail/group_span.rs:3:17
  |
3 | #[derive(Debug, Arbitrary)]
  |                 ^^^^^^^^^ expected `u32`, found `u8`
4 | struct X {
5 |     #[strategy((0u8..10u8))]
  |                 --------- this tail expression is of type `Range<u8>`
  |
note: required by a bound in `_strategy_of_x`
 --> tests/compile_fail/group_span.rs:6:8
  |
3 | #[derive(Debug, Arbitrary)]
  |                 --------- required by a bound in this function
...
6 |     x: u32,
  |        ^^^ required by this bound in `_strategy_of_x`
  = note: this error originates in the derive macro `Arbitrary` (in Nightly builds, run with -Z macro-backtrace for more info)
